lib/puppet/face/node/clean.rb

```diff
@@ -51,14 +45,23 @@ def cleanup(node)
     clean_reports(node)
   end
 
+  class LoggerIO
+    def err(message)
+      Puppet.err(message)
+    end
+
+    def inform(message)
+      Puppet.info(message)
+    end
+  end
+
   # clean signed cert for +host+
   def clean_cert(node)
-    if Puppet::SSL::CertificateAuthority.ca?
-      Puppet::Face[:ca, :current].revoke(node)
-      Puppet::Face[:ca, :current].destroy(node)
-      Puppet.info _("%{node} certificates removed from ca") % { node: node }
-    else
-      Puppet.info _("Not managing %{node} certs as this host is not a CA") % { node: node }
+    begin
+      require 'puppetserver/ca/cli'
+      Puppetserver::Ca::Action::Clean.new(LoggerIO.new).run({ 'certnames' => [node] })
```

Comment:
I do like calling the `Clean` action directly here instead of going through the `Cli` class like I was doing before, but the extra cruft that gets appended to the messages is pretty gross.
```
Error: Error:
Error:     Could not find files to clean for mycert
```
I could try to put more logic in the logger wrapper to strip the extra ones away.

Reply:
I like the way you've done the wrapping here, we could update the `err` method to not print the CLI's initial `Error:` line, like:
```rb
    def err(message)
      Puppet.err(message) unless message =~ /^\s*Error:\s*/
    end
```

You also mentioned the different logging levels. It looks like the Puppet CA logs at [notice](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/ssl/certificate_authority.rb#L296). Maybe we should set `inform` to wrap `Puppet.notice` rather than `Puppet.info` to more closely match the level of logging currently?

Reply:
Oh I like the regex thing. And yeah I think we want to preserver outputting something, the silence on success seemed weird. I'll update this.